daemon/oci_linux.go

```diff
@@ -739,7 +739,15 @@ func (daemon *Daemon) populateCommonSpec(s *specs.Spec, c *container.Container)
 	s.Process.Cwd = cwd
 	s.Process.Env = c.CreateDaemonEnvironment(c.Config.Tty, linkedEnv)
 	s.Process.Terminal = c.Config.Tty
-	s.Hostname = c.FullHostname()
+
+	s.Hostname = c.Config.Hostname
```

Comment:
I should point out that the original code (using `FullHostname` was not really correct before, since the `Domainname` was not properly exposed or used). So this part is not an accidental change.